traits/tests/test_callable.py

```diff
@@ -91,6 +118,21 @@ def test_callable_in_complex_trait(self):
                 a.callable_or_str = value
             self.assertEqual(a.callable_or_str, old_value)
 
+    def test_compound_callable_refcount(self):
+        # Regression test for enthought/traits#906.
+
+        def my_function():
+            pass
+
+        a = MyCallable()
+
+        string_value = "some string"
+        callable_value = my_function
+
+        for _ in range(10):
```

Comment:
Why is it 10 ?

Reply:
Hmm. No really good reason. Let me take out the loop and see if a single iteration is sufficient to cause the segfault.

Reply:
So a single iteration is not enough: on my own machine, I need at least 3. I've added a comment about the number of iterations.

My (unverified) conjecture is that what's happening is that on each iteration through the loop, `my_function` loses one of its references; when its reference count gets to zero, it gets garbage collected. Everything's fine unless we try to *access* `my_function` after it's been garbage collected. The second iteration is probably reducing the reference count to zero, and causing `my_function` to be garbage collected. The third iteration then accesses `my_function` and causes the segfault.